witness/src/Dscp/Witness/Web/Logic.hs

```diff
@@ -201,3 +204,19 @@ getHashType someHash = fmap (fromMaybe HashIsUnknown) . runMaybeT . asum $
     distinguishTx = \case
         GMoneyTx _ -> HashIsMoneyTx
         GPublicationTx _ -> HashIsPublicationTx
+
+-- | Check @'FairCV'@ against records in the public chain.
+checkFairCV :: forall ctx m. WitnessWorkMode ctx m => FairCV Unchecked -> m FairCVCheckResult
+checkFairCV =
+    either pure checkAgainstDB . validateFairCV
+  where
+    checkAgainstDB (FairCV cv) = FairCVCheckResult <$>
+        M.traverseWithKey (M.traverseWithKey . checkProofAgainstDB) cv
+    checkProofAgainstDB addr h proof =
+        maybe False (checkProofPure addr proof) <$>
```

Comment:
@Martoon-00 You're right, and `FairCVCheckResult` should be way more complex eventually - it should include not only the detailed type of failure in case of failure, but also assurance level in case of the success (i. e. the depth of the block in which the publication resides) But I intend to keep it simple in this PR and elaborate on the check result separately.